cryptoapi_windows.go

```diff
@@ -9,16 +9,66 @@ import (
 	"time"
 
 	"golang.org/x/sys/windows/registry"
+	"gopkg.in/hlandau/easyconfig.v1/cflag"
+)
+
+var (
+	cryptoApiFlagGroup            = cflag.NewGroup(flagGroup, "capi")
+	cryptoApiFlagLogicalStoreName = cflag.String(cryptoApiFlagGroup, "store", "Root",
+		"Name of CryptoAPI logical store to inject certificate into. Consider: Root, Trust, CA, My")
+	cryptoApiFlagStoreScope = cflag.String(cryptoApiFlagGroup, "scope", "system",
+		"Scope of CryptoAPI certificate store. Valid choices: current-user, system, enterprise, group-policy")
 )
 
-// In 64-bit Windows, this key is shared between 64-bit and 32-bit applications.
-// See https://msdn.microsoft.com/en-us/library/windows/desktop/aa384253.aspx
-const cryptoApiCertStoreRegistryBase = registry.LOCAL_MACHINE
-const cryptoApiCertStoreRegistryKey = `SOFTWARE\Microsoft\EnterpriseCertificates\Root\Certificates`
 const cryptoApiMagicName = "Namecoin"
 const cryptoApiMagicValue = 1
 
+var (
+	// cryptoApiStores consists of every implemented store.
+	// when adding a new one, the `%s` variable is optional.
+	// if `%s` exists in the Logical string, it is replaced with the value of -store flag
+	cryptoApiStores = map[string]Store{
+		"current-user": Store{registry.CURRENT_USER, `SOFTWARE\Microsoft\SystemCertificates\`, `%s\Certificates`},
+		"system":       Store{registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\SystemCertificates\`, `%s\Certificates`},
+		"enterprise":   Store{registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\EnterpriseCertificates\`, `%s\Certificates`},
+		"group-policy": Store{registry.LOCAL_MACHINE, `SOFTWARE\Policies\SystemCertificates\`, `%s\Certificates`},
```

Comment:
According to my notes, `SOFTWARE\Policies\SystemCertificates\` should be `SOFTWARE\Policies\Microsoft\SystemCertificates\`.  Not sure if this is a typo or if you found a different source that's not consistent with my notes.

Reply:
typo. good catch, thanks.

Reply:
Actually it may be `Policy\Microsoft` ?
( from https://docs.microsoft.com/en-us/windows/win32/seccrypto/system-store-locations#cert_system_current_user_group_policy )
```
HKEY_CURRENT_USER
   Software
      Policy
         Microsoft
            SystemCertificates
```

and 

```
HKEY_LOCAL_MACHINE
   Software
      Policy
         Microsoft
            SystemCertificates
```

Reply:
Confirmed on a windows machine that Policy does not exist, Policies is the one, fixed. :smile: